openeo/extra/job_management.py

```diff
@@ -202,6 +209,39 @@ def _normalize_df(self, df: pd.DataFrame) -> pd.DataFrame:
         df = df.assign(**new_columns)
 
         return df
+    
+    def _check_and_stop_long_running_jobs(self, df: pd.DataFrame):
+        """Check for long-running jobs and stop them if necessary."""
+        running_jobs = df.loc[
+            (df.status == "running")
+        
+        ]
+        for i in running_jobs.index:
+
+            try:
+                job_id = df.loc[i, "id"]
+                backend_name = df.loc[i, "backend_name"]
+                con = self._get_connection(backend_name)
+                the_job = con.job(job_id)
+
+                job_time_created = the_job.describe()['created']
+                job_time_created = datetime.datetime.strptime(job_time_created, '%Y-%m-%dT%H:%M:%SZ')
+                time_difference = (datetime.datetime.now() - job_time_created)
+
+                if (time_difference.total_seconds() > self.max_running_duration.total_seconds()):
```

Comment:
I think you can simplify all this date math a bit by just doing something like

```
if datetime.datetime.now(timezone.utc) > job_time_created + self.max_running_duration:
```

Reply:
done